### ISSUE TITLE
TypeHints.TypeHintDeclaration and traversableTypeHints

### ISSUE THREAD

**Issue by Fuco1** [slevomat/coding-standard#158]
Hi.

I'm not sure how to configure this.  I only got it working if I set fully qualified types in the codesniffer configuration and then use the same fully qualified type in the `@var` annotation.  Is this expected behaviour?  Shouldn't it infer the types based on the `use` declarations?  Surely that is possible because the `UnusedUses` sniff does the resolutions somehow.

**Comment by kukulich** [comment]
@Fuco1 Do you use last version? There was a bug that is fixed in last version.

**Comment by Fuco1** [comment]
I use 2.4.1, should I use master?

**Comment by kukulich** [comment]
No, dont use master, the bug was fixed in 2.4.1. Could you please post here your ruleset.xml and the code with traversable type hints?

**Comment by Fuco1** [comment]
The php code is 

```
<?php

namespace Entity\Renderers;

use Doctrine\ORM\Mapping as ORM;
use Entity;
use Fixtures;
use JsonSerializable;


/**
 * @ORM\Entity
 */
class RowTest extends Entity\Identified implements JsonSerializable, Fixtures\Serializable
{

	/**
	 * @ORM\OneToMany(targetEntity = "Item", mappedBy = "row", cascade = {"persist", "remove"})
	 * @ORM\OrderBy({"order" = "ASC"})
	 * @var Doctrine\Common\Collections\ArrayCollection
	 */
	protected $items;
}
```

Configuration

```xml
<?xml version="1.0"?>
<ruleset name="Standard">

    <rule ref="./vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/TypeHints/TypeHintDeclarationSniff.php" />
    <rule ref="SlevomatCodingStandard.TypeHints.TypeHintDeclaration">
      <properties>
        <property name="traversableTypeHints" type="array" value="
            \Doctrine\Common\Collections\ArrayCollection
        "/>
      </properties>
    </rule>

</ruleset>
```

When I use the `\Doctrine\...` type hint it properly highlights the error, but not with a relative type hint.

**Comment by kukulich** [comment]
Should be fixed in release 2.4.2.

For `UnusedUses` sniff see settings: https://github.com/slevomat/coding-standard#slevomatcodingstandardnamespacesunuseduses-

**Comment by Fuco1** [comment]
I confirm this works, thanks a lot for such a quick fix!
